src/js/components/Modal/Modal.tsx

```diff
@@ -59,6 +61,7 @@ const Modal: React.FC<Props> = ({
           <div className="u-sizeFill">
             <h2 data-testid="modal-heading">{heading}</h2>
           </div>
+          {customClose}
```

Comment:
Not sure if this is the best way to handle this. Basically just want a way to add a custom button in place of the showClose default "x"

Reply:
@saralohr what is the problem with not displaying the default close button and passing some elements to the header the way I did in staging?

If we have a way to provide a close button by passing it to the header (which accepts components and jsx), why do we need to add another option?

Reply:
The header element is an `h2` nested in a layout container already providing the structure for a close button via `showClose`. So if we pass to the `header` the markup gets pretty weird forcing us to nest a whole new layout container inside of the `h2`.

```
<div class="Modal-header u-flex u-flexJustifyBetween">
    <div class="u-sizeFill">
          <h2 data-testid="modal-heading">
           // this is where `header` populates
           // this is the content we'd have to pas into the h2 to have a header and a custom close button
            <div class="u-flex u-flexJustifyBetween u-flexAlignItemsCenter>
               <div>Header Content</div>
              <button>Save and Close</button>
           </div>
         </h2>
      </div>
      // this is where default close button lives and achieves the layout we want already
</div>
```

Reply:
I prefer the flexibility to have a dev create a layout rather in the header component than updating the modal with another close option @bradcypert can you comment on this please

Reply:
I can definitely see both preferences here, but I think I tend to lean towards Sara's implementation. Keeping the close icon separate from the "Header" makes a lot of sense to me, and it's probably way more accessible markup (you're not "closing the header" you're "closing the modal"). We'll want to keep the generated markup from TeamSnap UI as semantic as possible.

Reply:
@bradcypert thanks for the input.